src/main/java/be/minelabs/inventory/OrderedInventory.java

```diff
@@ -20,30 +23,12 @@ public OrderedInventory(ItemStack ... items) {
 
```

Comment:
Why not change the type to `NbtCompound` instead of `NbtList` ? It would simplefy the code.
That way you could just read and write without any of the crazy conversions.

That way the NBT would look like:
` ... , grid : { Items : [...]}, io_inv : { Items : [...] } ,...`

Instead of trying to force the "Items" key out by transforming them.